### PR TITLE
Add AliasCommand

### LINKED ISSUE #112: Add Alias Function
- Enable users to define their own aliases for commands

### PR DESCRIPTION
**Pull request by marc-97**
Added AliasCommand, test cases for relevant classes. Will update doc in separate PR

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FlashBackParser.java [hunk 1]
```diff
@@ -6,23 +6,10 @@
...
-import seedu.address.logic.commands.FindCommand;
-import seedu.address.logic.commands.HelpCommand;
-import seedu.address.logic.commands.ListCommand;
-import seedu.address.logic.commands.RedoCommand;
-import seedu.address.logic.commands.ReviewCommand;
-import seedu.address.logic.commands.SortCommand;
-import seedu.address.logic.commands.StatsCommand;
-import seedu.address.logic.commands.UndoCommand;
-import seedu.address.logic.commands.ViewCommand;
+import seedu.address.logic.commands.*;
```
Comment (linhns) [inline src/main/java/seedu/address/logic/parser/FlashBackParser.java]: I think wildcard import is not allowed

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -13,4 +13,7 @@
     public static final Prefix PREFIX_TAG = new Prefix("t/");
     public static final Prefix PREFIX_REMARK = new Prefix("r/");
     public static final Prefix PREFIX_FLAG = new Prefix("-");
+
+    public static final Prefix PREFIX_ALIAS_COMMAND = new Prefix("c/");
```
Comment (linhns) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: Should be a different prefix to `category`.

**Review by linhns — CHANGES_REQUESTED** [review]
Overall, your code looks pretty good. However, changes should be made towards CliSyntax to avoid repetition, Also, try to avoid wildcard import if possible

File: src/main/java/seedu/address/logic/parser/FlashBackParser.java (hunk 1 above)
Comment (marc-97) [reply src/main/java/seedu/address/logic/parser/FlashBackParser.java]: Thanks for the catch, will update accordingly

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 2 above)
Comment (marc-97) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java]: Understood. I updated the prefix for alias to be "cmd/" for command and "al/" for the alias name

File: src/main/java/seedu/address/logic/parser/AliasCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+public class AliasCommandParser implements Parser<AliasCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AliasCommand
+     * and returns an AliasCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AliasCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_ALIAS_COMMAND,
```
Comment (yeoutzer) [inline src/main/java/seedu/address/logic/parser/AliasCommandParser.java:26]: Could probably add in assertion or requireNonNull for defensive coding. Other than that, good job. Code looks neat and follow SLAP.

File: src/main/java/seedu/address/logic/commands/AliasCommand.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (yeoutzer) [inline src/main/java/seedu/address/logic/commands/AliasCommand.java]: Wild card import that you missed out when changing.

File: src/main/java/seedu/address/logic/parser/AliasCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (yeoutzer) [inline src/main/java/seedu/address/logic/parser/AliasCommandParser.java]: Wild card import as well.

File: src/main/java/seedu/address/model/AliasMap.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
+package seedu.address.model;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import seedu.address.logic.commands.*;
```
Comment (yeoutzer) [inline src/main/java/seedu/address/model/AliasMap.java]: Wild card import here as well.

**Review by yeoutzer — COMMENTED** [review]
Good job. Overall code looks good and neat. Might want to check through all the edited files again to see if there are any wildcard imports before merging.

**Pull request merged**